### ISSUE TITLE
Unable to upload using Slack.Web.Files api

### ISSUE THREAD

**Issue by bklang** [BlakeWilliams/Elixir-Slack#96]
According to the docs for [Slack.Web.Files.upload/3](https://hexdocs.pm/slack/Slack.Web.Files.html#upload/3), the first parameter to the upload function is the file, which will be sent as multipart/form-data. I've tried specifying a file, but the Slack API just response that no file data was sent:

```
iex(28)> Slack.Web.Files.upload "/tmp/test.jpg", "front_door.jpg", %{as_user: true, channels: ["@bklang"]}
%{"error" => "no_file_data", "ok" => false, "warning" => "superfluous_charset"}
```

I am able to use the `upload/3` function if I use the `:content` key, but then I can't specify a mime-type (I can only list one of the Slack-provided pseudo-types, like "php"), and am limited to just 1MB.   Am I doing something wrong, or is this a limitation of the auto-generated code?

**Comment by BlakeWilliams** [comment]
You're right, it seems to be a bit of the generated code and how HTTPoison handles multipart.

**Comment by BlakeWilliams** [comment]
I just pushed `0.9.2` which contains the fix from #97. Thanks for reporting this and let me know if you run into any more issues!

**Comment by bklang** [comment]
Wow that was quick! Thanks so much!
